crates/api_crud/src/site/update.rs

```diff
@@ -208,3 +183,246 @@ impl PerformCrud for EditSite {
     Ok(res)
   }
 }
+
+fn validate_update_payload(
+  local_site_slur_filter_regex: Option<String>,
+  federation_enabled: bool,
+  private_instance: bool,
+  edit_site: &EditSite,
+) -> LemmyResult<()> {
+  // Check that the slur regex compiles, and return the regex if valid...
+  // Prioritize using new slur regex from the request; if not provided, use the existing regex.
+  let slur_regex = build_and_check_regex(
+    &edit_site
+      .slur_filter_regex
+      .as_deref()
+      .or(local_site_slur_filter_regex.as_deref()),
+  )?;
+
+  if let Some(name) = &edit_site.name {
+    // The name doesn't need to be updated, but if provided it cannot be blanked out...
+    site_name_length_check(name)?;
+    check_slurs_opt(&edit_site.name, &slur_regex)?;
+  }
+
+  if let Some(desc) = &edit_site.description {
+    site_description_length_check(desc)?;
+    check_slurs_opt(&edit_site.description, &slur_regex)?;
+  }
+
+  if let Some(listing_type) = &edit_site.default_post_listing_type {
+    // Only allow all or local as default listing types...
+    if listing_type != &ListingType::All && listing_type != &ListingType::Local {
+      return Err(LemmyError::from_message(
+        "invalid_default_post_listing_type",
+      ));
+    }
+  }
+
+  let enabled_private_instance_with_federation = edit_site.private_instance == Some(true)
+    && edit_site.federation_enabled.unwrap_or(federation_enabled);
+  let enabled_federation_with_private_instance = edit_site.federation_enabled == Some(true)
+    && edit_site.private_instance.unwrap_or(private_instance);
+
+  if enabled_private_instance_with_federation || enabled_federation_with_private_instance {
+    return Err(LemmyError::from_message(
+      "cant_enable_private_instance_and_federation_together",
+    ));
+  }
+
+  // Ensure that the sidebar has fewer than the max num characters...
+  is_valid_body_field(&edit_site.sidebar)
+}
```

Comment:
You might be able to move a bunch of these checks into a common helper method to avoid duplication.